server/src/test/java/com/linecorp/line/auth/fido/fido2/server/bean/validation/mockmvc/AuthChallengeRequestBeanValidationTest.java

```diff
@@ -0,0 +1,61 @@
+package com.linecorp.line.auth.fido.fido2.server.bean.validation.mockmvc;
+
+import com.linecorp.line.auth.fido.fido2.common.server.AuthOptionRequest;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class AuthChallengeRequestBeanValidationTest extends BeanValidationTestSupport {
+
+    private MockMvc mockMvc;
+    private static AuthOptionRequest origin;
+    private AuthOptionRequest authChallengeRequest;
+
+    @BeforeAll
+    static void initGlobal() throws IOException {
+        origin = objectMapper.readValue(AuthOptionRequest.class.getResourceAsStream("/json/auth/auth-challenge-req.json"), AuthOptionRequest.class);
+    }
+
+    @BeforeEach
+    void setUp(final WebApplicationContext context) throws IOException {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(context)
+                .alwaysDo(MockMvcResultHandlers.print())
+                .build();
+
+        //Deep copy
+        authChallengeRequest = objectMapper.readValue(objectMapper.writeValueAsString(origin), AuthOptionRequest.class);
+    }
+
+    @Test
+    void validateSuccessfulRequest() throws Exception {
+
+        mockMvc.perform(post(AUTH_CHALLENGE_URL_PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(authChallengeRequest)))
+                .andExpect(status().isBadRequest());
+    }
```

Comment:
If the production code and the test code are packaged differently, we may not notice that there are tests.
How about writing them in `ChallengeControllerTest` as the same web layer test?

Reply:
Thank you for taking the time to give us your suggestions! As you mentioned, I've made modifications!